plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -174,5 +178,10 @@ private void validateCatalogs(List<CatalogMetadata> catalogs) {
     }
   }
 
-
+  // TODO: for now register storage engine functions here which should be static per storage engine
+  private void registerFunctions(String catalogName, StorageEngine storageEngine) {
```

Comment:
Can you add this method to the `CatalogService` interface?

Reply:
Could you explain why this need to be an API method of `CatalogService` interface? As I understand, storage-engine-defined functions are returned in `StorageEngine.getFunctions()` which represents additional capability of a storage engine. `CatalogService` can iterate static list of all storage engine (OpenSearch, Prometheus and S3 later) and register all functions internally.